pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller.go

```diff
@@ -174,6 +179,9 @@ func (n *klusterletController) sync(ctx context.Context, controllerContext facto
 		ExternalManagedKubeConfigWorkSecret:         helpers.ExternalManagedKubeConfigWork,
 		InstallMode:                                 klusterlet.Spec.DeployOption.Mode,
 		HubApiServerHostAlias:                       klusterlet.Spec.HubApiServerHostAlias,
+
+		RegistrationServiceAccount: serviceAccountName("registration-sa", klusterlet),
+		WorkServiceAccount:         serviceAccountName("work-sa", klusterlet),
```

Comment:
in InstallModeSingleton, the registration and work sa have the same name `%s-agent-sa` , currently it's ok because they have the same content in the yaml. but it may bring some confusion,  and in the future if someone change one of them, there may meet issue. 
could we consider to define a new sa yaml for Singleton mode?

Reply:
it will need to check the mode for each rolebinding and clusterrolebinding files. I think it is harder to maintain then just mutate the service account name.